client/src/containers/contact-us/index.tsx

```diff
@@ -0,0 +1,213 @@
+"use client";
+
+import { FC, FormEvent, useCallback, useRef, useState } from "react";
+
+import { useForm } from "react-hook-form";
+
+import Link from "next/link";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { ContactUsSchema } from "@shared/schemas/contact.schema";
+import { z } from "zod";
+
+import { client } from "@/lib/queryClient";
+
+import { PrivacyPolicySchema } from "@/containers/auth/signup/form";
+
+import { Button } from "@/components/ui/button";
+import { Checkbox } from "@/components/ui/checkbox";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Textarea } from "@/components/ui/textarea";
+import { toast } from "@/components/ui/use-toast";
+
+export const ContactUsWithPrivacyPolicySchema =
+  ContactUsSchema.merge(PrivacyPolicySchema);
+
+const ContactUsForm: FC = () => {
+  const [isSending, setIsSending] = useState<boolean>(false);
+  const formRef = useRef<HTMLFormElement>(null);
+  const form = useForm<z.infer<typeof ContactUsWithPrivacyPolicySchema>>({
+    resolver: zodResolver(ContactUsWithPrivacyPolicySchema),
+    // why omit privacyPolicy? The schema was instantiated with it.
+    defaultValues: {
+      name: "",
+      email: "",
+      message: "",
+      privacyPolicy: false,
+    },
+    mode: "onSubmit",
+  });
+
+  const handleContactUs = useCallback(
+    (evt: FormEvent<HTMLFormElement>) => {
+      evt.preventDefault();
+      setIsSending(true);
+
+      form.handleSubmit(async (formValues) => {
+        try {
+          const parsed = ContactUsWithPrivacyPolicySchema.omit({
+            privacyPolicy: true,
+          }).safeParse(formValues);
+          // todo: is this really required? the FormControl already validates the schema, and this is shared with the server, this might be redundant
+          if (parsed.success) {
+            await client.contact.contact.mutation({
+              body: parsed.data,
+            });
+
+            toast({ variant: "default", description: "Message sent!" });
```

Comment:
I think this toast will trigger regardless the mutation is successful or not, which might be a problem. I would makes sure with a `if (response.status === 200)` or similar than everything went well.

Reply:
Side note: you don't need to set the `variant` of the toast for `default`, this is already the default: https://github.com/Vizzuality/4-growth/blob/dev/client/src/components/ui/toast.tsx#L38-L40

Reply:
true, good catch